src/ioctls/vm.rs

```diff
@@ -610,6 +610,49 @@ impl VmFd {
         Ok(())
     }
 
+    /// Enable the specified capability as per the `KVM_ENABLE_CAP` ioctl.
+    ///
+    /// See the documentation for `KVM_ENABLE_CAP`.
+    ///
+    /// This ioctl returns Ok() when the capability is successfully enabled,
+    /// it returns an error otherwise.
+    ///
+    /// # Arguments
+    ///
+    /// * kvm_enable_cap - KVM capability structure. For details check the `kvm_enable_cap`
+    ///                    structure in the
+    ///                    [KVM API doc](https://www.kernel.org/doc/Documentation/virtual/kvm/api.txt).
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # extern crate kvm_ioctls;
+    /// extern crate kvm_bindings;
+    /// #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
```

Comment:
I am not a big fan of having x86_64 specific examples when the ioctl is available on other platforms as well. I would suggest to use `KVM_CAP_ARM_USER_IRQ` on aarch64 and `KVM_CAP_SPLIT_IRQCHIP` on x86_64. The only code that should be specific to the platform should be ```cap.cap = ...``` and ```cap.args[0] = 24```. Also add a comment about what is 24.

Reply:
Yes you're right, it's actually good to show an example with two different architectures.

Reply:
Well let me take that back... The cap `KVM_CAP_ARM_USER_IRQ` that you suggested is only meant to be checked through `check_extension()`, but it cannot be set from the VMM.
And based on the KVM bindings 4.20, there is no `arm/aarch64` capability that we could take as an example.
I reworked the structure of the documentation test though, let me know if it's okay with you.

Reply:
Yeah, I didn't look in depth. Is there really no other cap that we can enable for aarch64?

Reply:
Unless I'm reading KVM doc the wrong way (could be possible :p), I don't think there are some ARM caps that we can simply try to enable.
The only one could be `KVM_CAP_MANUAL_DIRTY_LOG_PROTECT2` but it's not part of our bindings because it's been introduced recently on the kernel.

Reply:
`KVM_CAP_MANUAL_DIRTY_LOG_PROTECT2` is only available starting with 5.0 so it will fail anyway since we don't have that capability on the host kernel.